components/raftstore/src/store/peer.rs

```diff
@@ -753,6 +756,13 @@ where
     ///   target peer.
     /// - all read requests must be rejected.
     pub pending_remove: bool,
+    /// Currently it's used to indicate whether the witness -> non-witess
+    /// convertion operation is complete. The meaning of completion is that
+    /// this peer must contain the applied data, then PD can consider that
+    /// the conversion operation is complete, and can continue to schedule
+    /// other operators to prevent the existence of multiple witnesses in
+    /// the same time period.
+    pub unavailable: Arc<AtomicBool>,
```

Comment:
why use `Arc`

Reply:
Because I need to notify the apply thread to modify the state after the apply snapshot is successful to process the committed message correctly. Is Arc<bool> expensive? Or pass it when I pass create ApplyTask ?

Reply:
pass it by ApplyRes